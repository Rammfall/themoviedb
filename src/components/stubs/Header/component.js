import React from 'react'
import {
  Typography, Row, Col, Avatar, Dropdown, Menu, Layout
} from 'antd'
import { CaretDownOutlined, UserOutlined } from '@ant-design/icons'
import { Link } from 'react-router-dom'

const Overlay = () => (
  <Menu>
    <Menu.Item>
      <Link to="/stubs/dashboard">Dashboard</Link>
    </Menu.Item>
    <Menu.Divider />
    <Menu.Item>
      <Link to="/stubs/lists">My Lists</Link>
    </Menu.Item>
    <Menu.Item>
      <Link to="/stubs/watchlist">Watchlist</Link>
    </Menu.Item>
    <Menu.Item>
      <Link to="/stubs/favorites">Favorites</Link>
    </Menu.Item>
    <Menu.Divider />
    <Menu.Item>Logout</Menu.Item>
  </Menu>
)

const Header = () => (
  <Layout.Header>
    <Row
      type="flex"
      justify="space-between"
    >
      <Col>
        <Typography.Text>THE MOVIE DB</Typography.Text>
      </Col>
      <Col>
        <Dropdown overlay={Overlay}>
          <Typography.Text>
            <Avatar icon={(<UserOutlined />)} />
            {' '}
            <span className="hide-sm-down">Username</span>
            {' '}
            <CaretDownOutlined />
          </Typography.Text>
        </Dropdown>
      </Col>
    </Row>
  </Layout.Header>
)

export default Header
