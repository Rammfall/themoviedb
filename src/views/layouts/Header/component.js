import React from 'react'
import { Avatar, Col, Dropdown, Layout, Row, Typography } from 'antd'
import { CaretDownOutlined, UserOutlined } from '@ant-design/icons'

import Menu from './Menu'

const HeaderComponent = () => (
  <Layout.Header>
    <Row
      type="flex"
      justify="space-between"
    >
      <Col>
        <Typography.Text>THE MOVIE DB</Typography.Text>
      </Col>
      <Col>
        <Dropdown overlay={<Menu />}>
          <Typography.Text>
            <Avatar icon={<UserOutlined />} />
            {' '}
            <span className="hide-sm-down">Username</span>
            {' '}
            <CaretDownOutlined />
          </Typography.Text>
        </Dropdown>
      </Col>
    </Row>
  </Layout.Header>
)

export default HeaderComponent
