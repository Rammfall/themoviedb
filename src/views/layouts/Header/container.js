import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { usernameSelector } from 'Store/concepts/account/selectors'

import { getUsername as getUsernameAction } from 'Store/concepts/account/actions'
import HeaderComponent from './component'

class Header extends Component {
  componentDidMount() {
    const { getUsername } = this.props

    getUsername()
  }

  render() {
    const { username } = this.props

    return (
      <HeaderComponent username={username} />
    )
  }
}

const mapStateToProps = (state) => ({
  username: usernameSelector(state)
})

const mapDispatchToProps = {
  getUsername: getUsernameAction
}

Header.propTypes = {
  username: PropTypes.string,
  getUsername: PropTypes.func.isRequired
}

Header.defaultProps = {
  username: null
}

export { HeaderComponent }
export default connect(mapStateToProps, mapDispatchToProps)(Header)
