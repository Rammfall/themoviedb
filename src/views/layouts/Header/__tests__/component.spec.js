import React from 'react'
import { shallow } from 'enzyme'

import Header from '../component'

describe('MenuComponent', () => {
  const defaultProps = {
    username: 'testUser'
  }

  it('matches snapshot', () => {
    const component = shallow(<Header {...defaultProps} />)

    expect(component).toMatchSnapshot()
  })
})
