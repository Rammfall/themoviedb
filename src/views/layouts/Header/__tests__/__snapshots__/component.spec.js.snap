// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuComponent matches snapshot 1`] = `
<Header>
  <Row
    justify="space-between"
    type="flex"
  >
    <Col>
      <Text>
        THE MOVIE DB
      </Text>
    </Col>
    <Col>
      <Dropdown
        mouseEnterDelay={0.15}
        mouseLeaveDelay={0.1}
        overlay={<Memo(Connect(Menu)) />}
      >
        <Text>
          <Avatar
            icon={<ForwardRef(UserOutlined) />}
            shape="circle"
            size="default"
          />
           
          <span
            className="hide-sm-down"
          >
            testUser
          </span>
           
          <ForwardRef(CaretDownOutlined) />
        </Text>
      </Dropdown>
    </Col>
  </Row>
</Header>
`;
